File: android/app/src/main/java/ai/ember/app/features/onboarding/OnboardingModels.kt
package ai.ember.app.features.onboarding

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * A single Q&A pair for the onboarding flow.
 *
 * Maps to the backend's `OnboardingAnswer` Pydantic model.
 */
@Serializable
data class OnboardingAnswer(
    @SerialName("question_key") val questionKey: String,
    val answer: String,
)

/**
 * Request body for POST /api/v1/onboarding/complete.
 */
@Serializable
data class OnboardingRequest(
    val answers: List<OnboardingAnswer>,
)

/**
 * Response from POST /api/v1/onboarding/complete.
 */
@Serializable
data class OnboardingResponse(
    @SerialName("onboarding_completed") val onboardingCompleted: Boolean,
    @SerialName("memories_seeded") val memoriesSeeded: Int,
)

/**
 * Represents a single onboarding question card.
 *
 * Pure domain model — no Android dependencies.
 */
data class OnboardingQuestion(
    val index: Int,
    val questionKey: String,
    val questionTextResId: Int,
    val placeholderResId: Int,
)

File: ios/Ember/Core/Models/OnboardingModels.swift
import Foundation

/// A single Q&A pair from the onboarding flow.
/// Maps to the backend's `OnboardingAnswer` Pydantic model.
struct OnboardingAnswer: Codable {
    let questionKey: String
    let answer: String
}

/// Request body for `POST /api/v1/onboarding/complete`.
/// Maps to the backend's `OnboardingRequest` Pydantic model.
struct OnboardingRequest: Codable {
    let answers: [OnboardingAnswer]
}

/// Response from `POST /api/v1/onboarding/complete`.
/// Maps to the backend's `OnboardingResponse` Pydantic model.
struct OnboardingResponse: Codable {
    let onboardingCompleted: Bool
    let memoriesSeeded: Int
}
